packages/core/src/middleware-http-auth-scheme/httpAuthSchemeMiddleware.ts

```diff
@@ -84,18 +88,20 @@ export const httpAuthSchemeMiddleware =
     const options = config.httpAuthSchemeProvider(
       await mwOptions.httpAuthSchemeParametersProvider(config, context as TContext, args.input)
     );
-    const authSchemes = convertHttpAuthSchemesToMap(config.httpAuthSchemes);
+    const optionsMap = convertHttpAuthOptionsToMap(options);
+
+    const authSchemePreference = await config.authSchemePreference();
```

Comment:
Add check for `config.authSchemePreference` for backward compatibility, since `@smithy` packages use `^`
```suggestion
    const authSchemePreference = config.authSchemePreference ? await config.authSchemePreference() : [];
```